sshd-core/src/main/java/org/apache/sshd/common/channel/ChannelOutputStream.java

```diff
@@ -42,6 +43,19 @@
  * @author <a href="mailto:dev@mina.apache.org">Apache MINA SSHD Project</a>
  */
 public class ChannelOutputStream extends OutputStream implements java.nio.channels.Channel, ChannelHolder {
+
+    private enum WriteState {
+        BUFFERED,
+        NEED_FLUSH,
+        NEED_SPACE
+    }
+
+    private enum OpenState {
```

Comment:
Let's make this *protected* (or even *public*) in the spirit of allowing users to overload anything they want in our code.

Reply:
Done.